simulation/sub8_simulation/sub8_sim_tools/physics/physics.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+'''Physics components for the Sub8 simulator, Simul8
+Original Author: Annie Luc
+'''
+from __future__ import division
+import traceback
+import numpy as np
+import time
+import sys
+import os
+import random
+import time
+import ode
+
+
+class Constants(object):
+    density_water = 999.97  # kg / m**3
+    g = -9.81  # m / s**2, gravitational constant
+
+
+class World(object):
+    def __init__(self, dt):
+        '''This section incorporates PyODE to define methods to create and draw three objects: a sphere, box, or cylinder.
+        Some code borrowed from http://sourceforge.net/p/pyode/mailman/message/19674697/.
+        '''
+        self.dt = dt
+        self.entities = []
+        # Create a world object
+        self.ode_world = ode.World()
+        self.ode_world.setGravity(np.array([0, 0, Constants.g]))    
+
+        self.ode_world.setERP(0.8)
+        self.ode_world.setCFM(1E-5)
+    
+        # Create a space object
+        self.space = ode.Space()
+    
+        # Create a plane geom which prevent the objects from falling forever
+        self.floor = ode.GeomPlane(self.space, (0., 0., 1.), -5.)
+        self.contact_group = ode.JointGroup()
+
+    def add_entity(self, Entity_Type, *args, **kwargs):
+        '''Add an arbitrary entity
+        Example:
+            Say we want to add a Sphere to my physics world
+            from physics import Sphere, World
+            w = World(dt=0.1)
+            w.add_entity(Sphere, postion, density, radius)
+
+        NOTE that we did not instantiate a Sphere, there was no Sphere(~~~)
+        We just passed the Sphere class, not a Sphere instance
+        This is for greater extensibility without repeated code
+        '''
+        entity = Entity_Type(self.ode_world, self.space, *args, **kwargs)
+        self.entities.append(entity)
+        return entity
+
+    def step(self, dt=None):
+        if dt is None:
+            dt = self.dt
+
+        # for i in range(2):
+        self.space.collide((self.ode_world, self.contact_group), self.near_callback)
+        self.ode_world.step(dt / 2)
+        for entity in self.entities:
+            entity.step(dt / 2)
+
+        # Remove all contact joints
+        self.contact_group.empty()
+
+    def near_callback(self, (world, contact_group), geom1, geom2):
+        '''Callback function for the collide() method.
+        This function checks if the given geoms do collide and
+        creates contact joints if they do.
+        '''
+        # Check if the objects do collide
+        contacts = ode.collide(geom1, geom2)
+        # Create contact joints
+        for c in contacts:
+            c.setBounce(.4)
+            c.setMu(5000)
+            j = ode.ContactJoint(world, contact_group, c)
+            j.attach(geom1.getBody(), geom2.getBody())
+
+
+class Entity(object):
+    _linear_damping_coeff = -0.2
```

Comment:
Do we want to be able to change these on a case by case basis,or is the desired functionality for these values to remain static?